packages/terra-action-header/src/_ActionHeaderContainer.jsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import styles from './ActionHeaderContainer.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * Child element to be displayed on the right end of the header.
+   * The element passed as children will be decorated with flex attributes.
+   */
+  children: PropTypes.element,
+
+  /**
+   * Content to be displayed at the start of the header, placed before the title.
+   */
+  startContent: PropTypes.element,
+
+  /**
+   * Text to be displayed as the title in the header bar.
+   */
+  title: PropTypes.string,
+
+  /**
+   * Content to be displayed at the end of the header.
+   * The element passed as endContent will be wrapped in a div with flex attributes.
+   */
+  endContent: PropTypes.element,
+  /**
+   * Sets the heading level. One of `1`, `2`, `3`, `4`, `5`, `6`.
+   */
+  level: PropTypes.oneOf([1, 2, 3, 4, 5, 6]).isRequired,
+};
+
+const defaultProps = {
+  title: undefined,
+  startContent: undefined,
+  endContent: undefined,
+};
+
+const ActionHeaderContainer = ({ children, title, startContent, endContent, level, ...customProps }) => {
+  const HeaderElement = `h${level}`;
+
+  return (
+    <header {...customProps} className={cx(['flex-header', customProps.className])}>
+      {startContent && <div className={cx('flex-end')}>{startContent}</div>}
+      <div className={cx('flex-fill')}>
+        {title &&
+          <div className={cx('title-container')}>
+            <HeaderElement className={cx('title')}>
+              {title}
+            </HeaderElement>
+          </div>
+        }
+      </div>
+      {children && React.cloneElement(children, { className: cx(['flex-collapse', children.props.className]) })}
```

Comment:
Does this work if multiple children are provided? Thinking we should map these with [React.Children](https://reactjs.org/docs/react-api.html#reactchildrenmap)

Reply:
Updated to use React.Children.map() [here](https://github.com/cerner/terra-core/commit/444fde0838557ce58e76c9faee9b854a6de8d62d).